components/conversations/CommentReactionPicker.js

```diff
@@ -91,18 +115,59 @@ const getOptimisticResponse = (comment, emoji, isAdding) => {
   }
 };
 
+const getOptimisticResponseForUpdates = (update, emoji, isAdding) => {
+  const userReactions = update.userReactions || [];
+  if (isAdding) {
+    const newCount = (update.reactions[emoji] || 0) + 1;
+    return {
+      __typename: 'Mutation',
+      addEmojiReaction: {
+        __typename: 'addEmojiReaction',
+        update: {
+          __typename: 'Update',
+          id: update.id,
+          reactions: { ...update.reactions, [emoji]: newCount },
+          userReactions: [...userReactions, emoji],
+        },
+      },
+    };
+  } else {
+    const newCount = (update.reactions[emoji] || 0) - 1;
+    const reactions = { ...update.reactions, [emoji]: newCount };
+
+    if (!reactions[emoji]) {
+      delete reactions[emoji];
+    }
+
+    return {
+      __typename: 'Mutation',
+      removeEmojiReaction: {
+        __typename: 'removeEmojiReaction',
+        update: {
+          __typename: 'Update',
+          id: update.id,
+          reactions,
+          userReactions: userReactions.filter(userEmoji => userEmoji !== emoji),
+        },
+      },
+    };
+  }
+};
+
 const mutationOptions = { context: API_V2_CONTEXT };
 
 /**
  * A component to render the reaction picker on comments.
  */
-const CommentReactionPicker = ({ comment }) => {
+const CommentReactionPicker = ({ comment, update }) => {
   const emojiFirstRow = ['👍️', '👎', '😀', '🎉'];
   const emojiSecondRow = ['😕', '❤️', '🚀', '👀'];
   const [open, setOpen] = React.useState(false);
   const wrapperRef = React.useRef();
   const [addCommentReaction] = useMutation(addCommentReactionMutation, mutationOptions);
   const [removeCommentReaction] = useMutation(removeCommentReactionMutation, mutationOptions);
+  const [addUpdateReaction] = useMutation(addReactionMutation, mutationOptions);
```

Comment:
Since the new mutations support both comments and updates, could we move everything there?

Reply:
@Betree : Yes, was thinking to do this in another PR. But I've just did it here. Thanks for pointing it out. 😄